app/controllers/weekly_questions_controller.rb

```diff
@@ -0,0 +1,23 @@
+class WeeklyQuestionsController < ApplicationController
+  before_action :authorize_request
+
+  # GET all the weekly questions with question body info
+  def index
+    weekly_questions = WeeklyQuestion.all
+
+    ret = weekly_questions.map do |week_question|
+      question = Question.find(week_question.question_id)
+      {
+        weekly_question: week_question,
+        question_body: question.body
+      }
+    end
+    render json: ret.to_json
```

Comment:
```suggestion
    render json: WeeklyQuestion.all.to_json(include: {question: {only: :body}})
```

If you do `weekly_question.question` ActiveRecord will do the find for you, since you have a `belongs_to` relation in the model, but also, you can tell `to_json` what you want and get rid of the `map` entirely.

It does change the structure of the json output so you need to change the FE as well, but it changes it for the better, honestly.

Reply:
I was trying to do this but didn't manage, thanks!